### DISCUSSION TITLE
Shebang line

### DISCUSSION THREAD

**Discussion by omerozyildirim** [thonny/thonny#3379] [General]
Is there a way to make Thonny respect shebang lines?

Shebang lines are the most practical way to utilize different interpreters for different files, especially with conda setups, which do not require activation once you define the correct interpreter on the shebang line. Besides, they are the exact mechanism the system uses when executing through CLI, making the IDE 100% compatible with the underlying OS.

Gemini includes Thonny in the list of IDEs that respect shebang lines at the time of this writing, although it cannot describe clearly how to activate it.

A single checkbox in the options window to use the shebang interpreter for the files, if any, synchronizing the shell interpreter with it would be an optimal solution.

**Comment by Hermann12** [comment]
Shebang lines respected on LINUX/ UNIX systems, only. This is for command line scripting, where you start the script outside of Thonny. Are you using a Linux system?

**Comment by omerozyildirim** [comment]
I use Kubuntu 24.10 and Thonny 4.0.1.

**Comment by aivarannamaa** [comment]
Thonny does not inspect the shebang line when running a script. It basically does `/path/to/python/python3 my_script.py` instead of running `python3` directly.
